Docs: add architecture diagram

GroenteLepel opened the pull request:
This commit adds an extensive diagram showing the architecture and workflow of the transformer thermal model. I opted for a state diagram instead of a class diagram or architecture diagram for a couple of reasons. Firstly, the architecture diagram is quite chaotic from Mermaid, and is mainly for architecture of web apps and stuff. The class
diagram is mainly focused on class inheritance, and also has no real flow to it.

My goal of this diagram was for someone to be able to visually read the diagram and understand the flow of the process, and what they needed to provide and where these items were used in the codebase.

I've added the necessary plugins so that it generates in MkDocs, and updated some theme colors to make everything readable and have a good overview. 

Some things that I'm not really sure how to handle yet:
 - No way to check this out online in a "test" branch, so you have to run it locally to check
 - This diagram is not linked to our code, so this is a manual dependency we need to keep up with. Luckily, this should only change for every major release we put out, so that wouldn't be too often.
 - Once this is accepted, since we do not have an automatic integration with our docs push, I will have to manually publish these docs.

Here is a screenshot of the page:
![image](https://github.com/user-attachments/assets/1abbab2a-12b1-4bf9-bf20-2c233942e899)

Harm1995 commented:
I think this is a nice addition to the documentation, and I agree that this state representation feels like the favorable choise. I have two small points:
- The thermal model also takes an initial top_oil temperature (and hotspot temperature, but this may be deleted in future versions) as input. This is optional, but I think it should still be included.
- To calculate the aging, we need the calculated hotspot temperatures. The output is also not added the the standard output schema, but a different type (Either a pandas series or a float). It therefore feels like the aging is better placed after the output of the thermal calculation output.

GroenteLepel commented:
Thanks, those are some valid points! I've added the input to the optional:

![image](https://github.com/user-attachments/assets/b182d7d2-cac9-4cc2-bcdf-f0bf4c5690a4)

And clarified that the `Model()` uses these for the initialisation:
![image](https://github.com/user-attachments/assets/c787e370-8955-4111-946b-75df1a29d571)

And finally I split the results of the aging:
![image](https://github.com/user-attachments/assets/f2fed5b9-2b71-49e1-912b-28273afb5b4b)

It now seems like the results are "bound together" into the `OutputProfile` at the final moment, where it officialy first creates an `OutputProfile`, and then uses that for the aging, but I found that this reads easier and looks like the "bundling of results" takes place waaay at the end, which I find more logical.

GroenteLepel commented:
I also updated the complete screenshot in the description of this PR
